public/directives/todoDetail/todoDetailDirective.js

```diff
@@ -1,16 +1,46 @@
-angular.module('app').directive('todoDetail', function() {
+angular.module('app').directive('todoDetail', function () {
 
-    return {
-        templateUrl: './directives/todoDetail/todoDetailTemplate.html',
-        replace: true,
-        controller: function($scope) {
+  return {
+    templateUrl: './directives/todoDetail/todoDetailTemplate.html',
+    replace: true,
+    controller: function ($scope) {
+      $scope.changesMade = false;
+      var originalVals = null;
 
-        },
-        scope: {
-            task: '<',
-            onSave: '&',
-            // onRestore: '&',
-            // onDelete: '&'
+      function resetValues() {
+        $scope.changesMade = false;
+        originalVals = angular.copy($scope.task);
+      }
+
+      $scope.$watch('task', function(newVal, oldVal) {
+        if(!angular.equals(newVal, oldVal)) {
+          originalVals = angular.copy(newVal);
+        }
+      });
```

Comment:
`$scope.$watch` listens for changes on whatever scope variable is named in the first parameter...in this case we're watching `$scope.task` since I passed in the string `'task'`. If the task object changes, then the callback function will be invoked. It will receive the old value of task, and the new value of task as arguments.